browser/models.py

```diff
@@ -212,22 +217,26 @@ def get(project_id):
         return project
 
     @staticmethod
-    def create(filename, input_bucket, output_bucket, path):
+    def create(path, bucket):
         """
         Create a new project.
         Wraps the Project constructor with logging and database commits.
 
         Args:
-            filename (str): filename including .npz or .trk extension
-            input_bucket (str): S3 bucket to download file
-            output_bucket (str): S3 bucket to upload file
             path (str): full path to download & upload file in buckets; includes filename
+            bucket (str): S3 bucket to download file
 
         Returns:
             Project: new row in the Project table
         """
         start = timeit.default_timer()
-        new_project = Project(filename, input_bucket, output_bucket, path)
+        new_project = Project(path, bucket)
+        # Assign a unique 12 character base64 token to the project
+        while True:
+            token = token_urlsafe(9)  # 9 bytes is 12 base64 characters
+            if not db.session.query(Project).filter_by(token=token).first():
+                new_project.token = token
+                break
```

Comment:
Just keep trying to make a random token until we find one that is unused?

Also, is there a specific reason you chose 12 chars long?

Reply:
Yes, keep making random ones until it's unused. This should happen immediately, as the probability of a collision when we have ~60,000 projects (which I think is how are in the current db) and use 12 character IDs is `2^16 / (2^8)^12` = 2^(-80)`.

The number of base 64 characters needed to represent `n` bytes is `n * 4 / 3`, so I wanted to choose a multiple of 3, and youtube uses 11 characters, so I chose 12.